firmware/application/irq_controls.cpp

```diff
@@ -122,25 +122,56 @@ static bool touch_update() {
 
 static uint8_t switches_raw = 0;
 
+/* Type to access the bits in the raw switch data.
+ * The raw data is not packed in a way that makes looping over the
+ * data easy. One option is this reader, the other would be to swizzle
+ * the bits to make the order work. */
+struct RawSwitch {
+    const uint8_t raw_{0};
+
+    uint8_t right() const { return (raw_ >> 0) & 1; }
+    uint8_t left() const { return (raw_ >> 1) & 1; }
+    uint8_t down() const { return (raw_ >> 2) & 1; }
+    uint8_t up() const { return (raw_ >> 3) & 1; }
+    uint8_t select() const { return (raw_ >> 4) & 1; }
+    uint8_t rot_a() const { return (raw_ >> 5) & 1; }
+    uint8_t rot_b() const { return (raw_ >> 6) & 1; }
+    uint8_t dfu() const { return (raw_ >> 7) & 1; }
+};
+
 static bool switches_update(const uint8_t raw) {
     // TODO: Only fire event on press, not release?
     bool switch_changed = false;
-    for (size_t i = 0; i < switch_debounce.size(); i++) {
-        switch_changed |= switch_debounce[i].feed((raw >> i) & 1);
-    }
+    auto r = RawSwitch{raw};
+
+    switch_changed |= switch_debounce[toUType(Switch::Right)].feed(r.right());
```

Comment:
Should I keep this or just swizzle the raw value? Hard to say which will be smaller code.